resources/js/features/user-game-activity/components/UserGameActivityTimeline/UserGameActivityTimeline.tsx

```diff
@@ -0,0 +1,154 @@
+import dayjs from 'dayjs';
+import utc from 'dayjs/plugin/utc';
+import type { FC } from 'react';
+import { FaFileArchive } from 'react-icons/fa';
+
+import { Timeline, TimelineItem } from '@/common/components/Timeline';
+import { usePageProps } from '@/common/hooks/usePageProps';
+import { cn } from '@/common/utils/cn';
+import { formatDate } from '@/common/utils/l10n/formatDate';
+
+import { getShouldAchievementSessionBeVisible } from '../../utils/getShouldAchievementSessionBeVisible';
+import { ClientLabel } from './ClientLabel';
+import { HashLabel } from './HashLabel';
+import { PlaytimeLabel } from './PlaytimeLabel';
+import { SessionTimelineEvent } from './SessionTimelineEvent';
+
+dayjs.extend(utc);
+
+interface UserGameActivityTimelineProps {
+  isOnlyShowingAchievementSessions: boolean;
+}
+
+export const UserGameActivityTimeline: FC<UserGameActivityTimelineProps> = ({
+  isOnlyShowingAchievementSessions,
+}) => {
+  const { activity } = usePageProps<App.Platform.Data.PlayerGameActivityPageProps>();
+
+  const { sessions } = activity;
+
+  // If it's a manual unlock session but there are no events in the session,
+  // then there's nothing we can show.
+  const filteredSessions = sessions.filter((s) => {
+    if (s.type === 'manual-unlock' && !s.events.length) {
+      return false;
+    }
+
+    return true;
+  });
```

Comment:
This change resolves:
> This page comes up blank: http://localhost:64000/user/SporyTike/game/25657/activity2

It looks like when a lot of manual unlocks happen for a user, one of those back-end services is creating numerous empty sessions for the unlocks.

I'm kind of afraid to touch some of that logic - rather than messing with the services, I just filter out those "dead" sessions.